src/service/resources-impl.js

```diff
@@ -784,9 +793,16 @@ export class Resources {
             minTop = minTop == -1 ? box.top : Math.min(minTop, box.top);
           }
           request.resource./*OK*/changeSize(
-              request.newHeight, request.newWidth, request.callback);
+              request.newHeight, request.newWidth);
+          if (request.callback) {
+            request.callback(/* hasSizeChanged */true);
+          }
           request.resource.overflowCallback(/* overflown */ false,
               request.newHeight, request.newWidth);
+        } else {
```

Comment:
Instead of more conditionals, we know but the end here that `resize` is either `true` (we did resize) or `false` (we will not resize). So we can move every `request.callback(X)` inside here to just one call, at the very end, of `request.callback(resize)`.